src/components/CommentForm.jsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { postComment } from "../api";
+import { ConstructionOutlined, ContactPageSharp } from "@mui/icons-material";
+
+const CommentForm = ({
+  article_id,
+  onCancel,
+  userList,
+  commentCount,
+  setCommentCount,
+  setIsCommenting,
+}) => {
+  const [username, setUsername] = useState("");
+  const [comment, setComment] = useState("");
+  const [isExpanded, setExpanded] = useState(false);
+  const [isPostingComment, setIsPostingComment] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+  const [loadingDone, setLoadingDone] = useState(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setIsPostingComment(true);
+    setLoadingDone(false);
+
+    postComment(article_id, username, comment).then((result) => {
+      if (result.status === 201) {
+        setUsername("");
+        setComment("");
+        setLoadingDone(true);
+        setIsPostingComment(false);
+        setCommentCount(commentCount + 1);
+        setIsCommenting(false);
+      }
+    });
+  };
+  const handleValidation = () => {
+    !isExpanded ? setIsValid(false) : setIsValid(true);
+  };
+  const handleExpand = () => {
+    const userListArr = userList.map((user) => {
+      return user.author;
+    });
+    const isAValidUserName = userListArr.includes(username);
+    isAValidUserName
+      ? setIsValid(true) & setExpanded(true)
+      : setExpanded(false);
+  };
+  return (
+    <div className="overlay">
+      <form onSubmit={handleSubmit} className="edit-card" id="overlay-content">
+        <label htmlFor="username">
+          {isValid ? "Username" : "Invalid Username!"}
```

Comment:
Just how is anyone using this app meant to know a valid username? I would highly recommend keeping a user in either a context or hard coded for a better user experience.